distributed/protocol/core.py

```diff
@@ -176,6 +176,7 @@ def loads_msgpack(header, payload):
     See Also:
         dumps_msgpack
     """
+    header = bytes(header)
     if header:
         header = msgpack.loads(header, use_list=False, **msgpack_opts)
```

Comment:
Do you recall why this was needed? Was this due to the `if header` line? Did `msgpack.dumps` need this? Or was it due to something else like potentially unusual types being passed in for `header` (like maybe a NumPy array)?

Reply:
Ah I guess this is explained here ( https://github.com/dask/distributed/commit/44c1d5cc6a867dcaa055518b6c7615863f498ebd ).